vcd/resource_vcd_org_vdc_access_control.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"strings"
+)
+
+func resourceVcdOrgVdcAccessControl() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVdcVdcAccessControlCreateUpdate,
+		ReadContext:   resourceVdcVdcAccessControlRead,
+		UpdateContext: resourceVdcVdcAccessControlCreateUpdate,
+		DeleteContext: resourceVdcVdcAccessControlDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVdcVdcAccessControlImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"shared_with_everyone": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the VDC is shared with everyone",
+			},
+			"everyone_access_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+				Description:  "Access level when the VDC is shared with everyone (only ReadOnly is available). Required when shared_with_everyone is set",
+			},
+			"shared_with": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"user_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the user to which we are sharing. Required if group_id is not set",
+						},
+						"group_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the group to which we are sharing. Required if user_id is not set",
+						},
+						"subject_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the subject (group or user) with which we are sharing",
+						},
+						"access_level": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+							Description:  "The access level for the user or group to which we are sharing. (Only ReadOnly is available)",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVdcVdcAccessControlCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	var accessControl types.ControlAccessParams
```

Comment:
do we need a lock for VDC - in case VDC is updated and access control changed, as VCD tend to expect fail when resource is busy

Reply:
We do not need it. Let me explain:
I created a script that was able to send the PUT request 2-3 times every second. I let it running. I started creating/changing/deleting VDC things like networks, edge gateways and named disks and the PUT request never failed.